WordPress/src/main/java/org/wordpress/android/util/PerAppLocaleManager.kt

```diff
@@ -118,7 +124,30 @@ class PerAppLocaleManager @Inject constructor(
         }
     }
 
-    companion object {
-        const val EXPERIMENTAL_PER_APP_LANGUAGE_PREF_KEY = "experimental_per_app_language_prefs"
+    /**
+     * Called when the device language is changed from our in-app language picker
+     * TODO Detect when language changed from app settings dialog
+     */
+    fun onLanguageChanged(languageCode: String) {
+        if (languageCode.isEmpty()) {
+            return
+        }
+
+        // Only update if the language is different
+        if (languageCode.equals(getCurrentLocaleLanguageCode()).not()) {
+            setCurrentLocaleByLanguageCode(languageCode)
+        }
+
```

Comment:
Question (❓): During transition, this below logic got completely removed from this function, can you help me understand why? 🙏 

```
        LocaleManager.setNewLocale(WordPress.getContext(), languageCode);
        WordPress.updateContextLocale();
        mContextProvider.refreshContext();
```

Reply:
This and the following are no longer necessary since the system now takes care of applying the per-app language. In a subsequent PR I'll be removing quite a bit of logic from `LocaleManager` due to this.

Reply:
Great, thanks for verifying that for me Nick, appreciated! 🙇